### PR TITLE
cmake: disable backtraces on AArch64 Linux

### LINKED ISSUE #8572: libunwind segfault when preparing crash report on AArch64 Linux
**Bug description**

Crash report is not printed on AArch64 Linux.
On x86_64 Linux and AArch64 macOS it works fine.

```
Tarantool 3.0.0-entrypoint-177-g199e20b9f
Target: Linux-aarch64-Debug
Build options: cmake . -DCMAKE_INSTALL_PREFIX=/usr/local -DENABLE_BACKTRACE=TRUE
Compiler: GNU-9.4.0
C_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -fopenmp -fmacro-prefix-map=/home/ubuntu/tarantool=. -std=c11 -Wall -Wextra -Wno-gnu-alignof-expression -fno-gnu89-inline -Wno-cast-function-type -Werror
CXX_FLAGS: -fexceptions -funwind-tables -fasynchronous-unwind-tables -fno-common -fopenmp -fmacro-prefix-map=/home/ubuntu/tarantool=. -std=c++11 -Wall -Wextra -Wno-invalid-offsetof -Wno-gnu-alignof-expression -Wno-cast-function-type -Werror
```

`2.11.0-rc1-108-g8862fb2fc` is also affected,
however `2.10.7-entrypoint-17-gbe2ad94ff` is not.

Reproducible on:
```
Ubuntu 20.04.3 LTS
Linux ip-172-31-28-148 5.15.0-1015-aws #19~20.04.1-Ubuntu SMP Wed Jun 22 19:10:51 UTC 2022 aarch64 aarch64 aarch64 GNU/Linux
```

Unable to reproduce on:
```
Ubuntu 18.04.5 LTS
Linux odroid 4.9.241-77 #1 SMP PREEMPT Mon Mar 29 23:04:14 -03 2021 aarch64 aarch64 aarch64 GNU/Linux
```

**Steps to reproduce**

1. Run a Tarantool instance;
1. Send a fatal signal, e.g.:
```
$ killall -SIGILL tarantool
```

**Actual behavior**

```
Tarantool 3.0.0-entrypoint-177-g199e20b9f
type 'help' for interactive help
tarantool> Fatal 11 while backtracing
Aborted (core dumped)
```

**Expected behavior**

```
Tarantool 3.0.0-entrypoint-177-g199e20b9f
type 'help' for interactive help
tarantool> Illegal instruction
  code: 0
  addr: 0x3e8002ca873
  context: 0xffffc4718ab0
  siginfo: 0xffffc4718a30
Current time: 1681487979
Please file a bug at http://github.com/tarantool/tarantool/issues
Attempting backtrace... Note: since the server has already crashed,
this may fail as well
#1  0xaaaab447cbfc in crash_collect+208
#2  0xaaaab447cf68 in crash_signal_cb+92
#3  0xffffa050e78c in ??+0
#4  0xffff9f8661b4 in epoll_pwait+60
#5  0xaaaab47d7e7c in epoll_poll+176
#6  0xaaaab47dc8bc in ev_run+1252
#7  0xaaaab44581e4 in tarantool_lua_run_script+392
#8  0xaaaab4215624 in main+1924
#9  0xffff9f7b6d50 in __libc_start_main+232
#10 0xaaaab4213144 in _start+52
#11 0xaaaab4213144 in _start+52
Aborted (core dumped)
```

**Backtrace**

```
Program received signal SIGSEGV, Segmentation fault.
_ULaarch64_tdep_trace (cursor=cursor@entry=0xffffffffca98, buffer=buffer@entry=0xffffffffdb48, size=size@entry=0xffffffffc6a4) at third_party/libunwind/src/aarch64/Gtrace.c:486
486	        ACCESS_MEM_FAST(ret, c->validate, d, cfa + f->lr_cfa_offset, pc);
(gdb) bt
#0  _ULaarch64_tdep_trace (cursor=cursor@entry=0xffffffffca98, buffer=buffer@entry=0xffffffffdb48, size=size@entry=0xffffffffc6a4) at third_party/libunwind/src/aarch64/Gtrace.c:486
#1  0x0000aaaaab0e46ec in unw_backtrace (buffer=0xffffffffdb48, size=64) at third_party/libunwind/src/mi/backtrace.c:69
#2  0x0000aaaaaad80dd0 in collect_current_stack (bt=0xffffffffdb40, stack=0x0) at src/lib/core/backtrace.c:50
#3  0x0000aaaaaad80ed0 in backtrace_collect (bt=0xffffffffdb40, fiber=0x0, skip_frames=2) at src/lib/core/backtrace.c:106
#4  0x0000aaaaaad70084 in crash_collect (signo=8, siginfo=0xffffffffde30, ucontext=0xffffffffdeb0) at src/lib/core/crash.c:45
#5  0x0000aaaaaad703f0 in crash_signal_cb (signo=8, siginfo=0xffffffffde30, context=0xffffffffdeb0) at src/lib/core/crash.c:183
#6  <signal handler called>
```

or

```
Thread 1 "tarantool" received signal SIGSEGV, Segmentation fault.
apply_reg_state (c=c@entry=0xffffffffca98, rs=rs@entry=0xffffffffc1e0) at third_party/libunwind/src/dwarf/Gparser.c:952
952	    ret = dwarf_get (c, c->loc[rs->ret_addr_column], &ip);
(gdb) bt
#0  apply_reg_state (c=c@entry=0xffffffffca98, rs=rs@entry=0xffffffffc1e0) at third_party/libunwind/src/dwarf/Gparser.c:952
#1  0x0000aaaaab0e71f8 in _ULaarch64_dwarf_step (c=c@entry=0xffffffffca98) at third_party/libunwind/src/dwarf/Gparser.c:1054
#2  0x0000aaaaab0e4fd0 in _ULaarch64_step (cursor=cursor@entry=0xffffffffca98) at third_party/libunwind/src/aarch64/Gstep.c:162
#3  0x0000aaaaab0e5890 in trace_init_addr (sp=<optimized out>, fp=<optimized out>, pc=<optimized out>, cfa=<optimized out>, cursor=0xffffffffca98, f=0xfffff4d74640) at third_party/libunwind/include/tdep-aarch64/libunwind_i.h:161
#4  trace_lookup (sp=<optimized out>, fp=<optimized out>, pc=<optimized out>, cfa=<optimized out>, cache=<optimized out>, cursor=0xffffffffca98) at third_party/libunwind/src/aarch64/Gtrace.c:333
#5  _ULaarch64_tdep_trace (cursor=cursor@entry=0xffffffffca98, buffer=buffer@entry=0xffffffffdb48, size=size@entry=0xffffffffc6a4) at third_party/libunwind/src/aarch64/Gtrace.c:451
#6  0x0000aaaaab0e46ec in unw_backtrace (buffer=0xffffffffdb48, size=64) at third_party/libunwind/src/mi/backtrace.c:69
#7  0x0000aaaaaad80dd0 in collect_current_stack (bt=0xffffffffdb40, stack=0x0) at src/lib/core/backtrace.c:50
#8  0x0000aaaaaad80ed0 in backtrace_collect (bt=0xffffffffdb40, fiber=0x0, skip_frames=2) at src/lib/core/backtrace.c:106
#9  0x0000aaaaaad70084 in crash_collect (signo=4, siginfo=0xffffffffde30, ucontext=0xffffffffdeb0) at src/lib/core/crash.c:45
#10 0x0000aaaaaad703f0 in crash_signal_cb (signo=4, siginfo=0xffffffffde30, context=0xffffffffdeb0) at src/lib/core/crash.c:183
#11 <signal handler called>
```

### PR DESCRIPTION
**Pull request by Gumix**
**build: remove backtrace feature compiler dependency from rpm spec**

The ability to support backtraces is checked in `cmake/compiler.cmake`, it makes no sense to duplicate the check in `rpm/tarantool.spec`. Also do not enable backtraces unconditionally in `apk/APKBUILD` and `static-build`.

Part of https://github.com/tarantool/tarantool/issues/6998

---

**build: disable backtrace feature on AArch64 Linux**

There are sporadic segfaults in libunwind during `backtrace_collect()`. Reproducible with the latest version, and there are open issues with similar stacks:

* https://github.com/libunwind/libunwind/issues/150
* https://github.com/libunwind/libunwind/issues/260
* https://github.com/libunwind/libunwind/issues/473

Let's disable `ENABLE_BACKTRACE` for AArch64 Linux until these issues are resolved in libunwind.

Closes #8572
Part of #8791

### CODE REVIEW

File: changelogs/unreleased/gh-8572-disable-backtraces-on-aarch64-linux.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+## bugfix/core
+
+* Disabled backtrace collection on AArch64 Linux (gh-8572).
```
Comment (CuriousGeorgiy) [inline changelogs/unreleased/gh-8572-disable-backtraces-on-aarch64-linux.md:3]:
```suggestion
* Disabled the backtrace collection feature on the AArch64 Linux platform (gh-8572).
```

**Review by CuriousGeorgiy — APPROVED** [review]

**Review by andreyaksenov — APPROVED** [review]

**Review by Totktonada — APPROVED** [review]
No objections from my side.

File: changelogs/unreleased/gh-8572-disable-backtraces-on-aarch64-linux.md (hunk 1 above)
Comment (Gumix) [reply changelogs/unreleased/gh-8572-disable-backtraces-on-aarch64-linux.md]: Fixed.

**Review by locker — APPROVED** [review]

**Comment by locker** [comment]
Cherry-picked to 2.11.

**Comment by azat** [comment]
The problem is likely due to very small stack size for sigaltstack, since aarch64 has lots of registers

**Pull request merged**